public/editor/scripts/project/project.js

```diff
@@ -237,22 +240,31 @@ define(function(require) {
             return callback(err);
           }
 
-          // Find an HTML file to open in the project, hopefully /index.html
-          var sh = new _fs.Shell();
-          sh.find(getRoot(), {name: "*.html"}, function(err, found) {
-            if(err) {
-              return callback(err);
+          // Find the index.html file in the project root to open
+          var indexLocation = Path.join(getRoot(), "index.html");
+          _fs.exists(indexLocation, function(exists) {
+            if(exists) {
+              callback(null, indexLocation);
+              return;
             }
-
-            // Look for an HTML file to open, ideally index.html
-            var indexPos = 0;
-            found.forEach(function(path, idx) {
-              if(Path.basename(path) === "index.html") {
-                indexPos = idx;
+            // Create a default index.html file
+            $.get(DEFAULT_INDEX_HTML_URL).then(function(data) {
+              _fs.writeFile(indexLocation, data, function(err) {
+                if (err) {
+                  console.error("Cannot write file to project: " + err);
+                  callback(err);
+                  return;
+                }
+                callback(null, indexLocation);
+              });
+            }, function(err) {
```

Comment:
there's no need to log here because:
1. The log doesn't really show anything too useful.
2. The browser will automatically log network failures.

Instead of this function, I would just pass `callback` here directly.